e2e/python/resolve_dependencies.spec.ts

```diff
@@ -42,62 +42,69 @@ if (
     "--db-source mysql+pymysql://user:pass@localhost:3306/mydb",
   ];
 
+  const observabilityOptions = ["llamatrace", "traceloop"];
+
+  // Run separate tests for each observability option to reduce CI runtime
+  test.describe("Test resolve python dependencies with observability", () => {
+    // Testing with streaming template, vectorDb: none, tools: none, and dataSource: --example-file
+    for (const observability of observabilityOptions) {
+      test(`observability: ${observability}`, async () => {
+        const cwd = await createTestDir();
+
+        await createAndCheckLlamaProject({
+          options: {
+            cwd,
+            templateType: "streaming",
+            templateFramework,
+            dataSource,
+            vectorDb: "none",
+            tools: "none",
+            port: 3000, // port, not used
+            externalPort: 8000, // externalPort, not used
+            postInstallAction: "none", // postInstallAction
+            templateUI: undefined, // ui
+            appType: "--no-frontend", // appType
+            llamaCloudProjectName: undefined, // llamaCloudProjectName
+            llamaCloudIndexName: undefined, // llamaCloudIndexName
+            observability,
+          },
+        });
+      });
+    }
+  });
+
   test.describe("Test resolve python dependencies", () => {
     for (const vectorDb of vectorDbs) {
       for (const tool of toolOptions) {
         for (const dataSource of dataSources) {
```

Comment:
@leehuwuj how about we run not each combination but just first all vec dbs, then all tools and then all DS?